src/cli.js

```diff
@@ -1,103 +1,196 @@
 #!/usr/bin/env node
 
-'use strict';
-
 import path from 'path';
 import chokidar from 'chokidar';
+import fs from 'fs';
 import glob from 'glob';
 import yargs from 'yargs';
 import chalk from 'chalk';
 import DtsCreator from 'typed-css-modules';
 import sass from 'node-sass';
 
+const pkg = require('../package.json');
 
-let yarg = yargs.usage('Create .css.d.ts from CSS modules *.css files.\nUsage: $0 [options] <input directory>')
-    .example('$0 src/styles')
-    .example('$0 src -o dist')
-    .example('$0 -p styles/**/*.icss -w')
-    .detectLocale(false)
-    .demand(['_'])
-    .alias('c', 'camelCase').describe('c', 'Convert CSS class tokens to camelcase')
-    .alias('o', 'outDir').describe('o', 'Output directory')
-    .alias('p', 'pattern').describe('p', 'Glob pattern with css files')
-    .alias('w', 'watch').describe('w', 'Watch input directory\'s css files or pattern').boolean('w')
-    .alias('d', 'dropExtension').describe('d', 'Drop the input files extension').boolean('d')
-    .alias('h', 'help').help('h')
-    .version(() => require('../package.json').version)
-let argv = yarg.argv;
-let creator;
-
-function writeFile(f) {
-    getSource(f).then((content) => {
-        creator.create(f, content, !!argv.w)
-        .then(content => content.writeFile())
-        .then(content => {
-            console.log('Wrote ' + chalk.green(content.outputFilePath));
-            content.messageList.forEach(message => {
-                console.warn(chalk.yellow('[Warn] ' + message));
-            });
-        })
-        .catch(reason => console.error(chalk.red('[Error] ' + reason)));
-    });
-};
+const readSass = (file, relativeTo) => (
+    new Promise((resolve, reject) => {
+        sass.render(
+            { file },
+            (err, result) => {
+                if (err && (relativeTo && relativeTo !== '/')) {
+                    return resolve([]);
+                } else if (err && (!relativeTo || relativeTo === '/')) {
+                    return reject(err);
+                }
+                return resolve(result.css.toString());
+            },
+        );
+    })
+);
 
-function getSource(file, relativeTo) {
-    return new Promise((resolve, reject) => {
-      sass.render({file}, (err, result) => {
-        if (err && relativeTo && relativeTo !== '/') {
-          return resolve([]);
+const createTypings = (f, creator, cache, handleError, handleWarning, verbose) => (
+    readSass(f)
+    .catch((reason) => {
```

Comment:
Does it create an empty file when error happens in readSass? I think the process should just fail and terminate, the catch block at the end of the chain can handle and report it.

Reply:
The catch block doesn't return anything, so the promise chain should close after the catch

Reply:
That is not true. Not returning anything does not terminate the chain. The subsequent then will get undefined. And It would be a problem because it will try to create a file with undefined content.

Reply:
Looks like I was mistaken. Good catch!